frontend/dockerfile/linter/docs/InvalidDefaultArgInFrom.md

```diff
@@ -0,0 +1,32 @@
+## Output
+
+```text
+Using the global ARGs with default values should produce a valid build.
+```
+
+## Description
+
+An `ARG` used in an image reference should be valid when no build arguments are used. An image build should not require `--build-arg` to be used to produce a valid build.
+
+## Examples
+
+❌ Bad: don't rely on an ARG being set for an image reference to be valid
+
+```dockerfile
+ARG TAG
+FROM busybox:${TAG}
+```
```

Comment:
Good (not great but no warning)

Reply:
I think we could add it as a "good" example then

Reply:
Well, we shouldn't recommend writing it like this.

```
ARG TAG=3.14
FROM alpine:${TAG}
```

^ is better(because for example now the default value shows up in `--call=outline`) but your version is valid as well (as it does not fail the build by default).

Reply:
https://github.com/moby/buildkit/pull/5113/commits/18578ef82cfa9903b4c967624e4da028e733255f